pallets/parachain-staking/src/lib.rs

```diff
@@ -1652,26 +1652,87 @@ pub mod pallet {
 		}
 	}
 
-	/// Add reward points to block authors:
-	/// * 20 points to the block producer for producing a block in the chain
-	impl<T: Config> nimbus_primitives::EventHandler<T::AccountId> for Pallet<T> {
+	impl<T: Config> Get<Vec<T::AccountId>> for Pallet<T> {
+		fn get() -> Vec<T::AccountId> {
+			Self::selected_candidates()
+		}
+	}
+
+	impl<T: Config> pallet_authorship::EventHandler<T::AccountId, T::BlockNumber> for Pallet<T> {
+		// Add reward points to block authors:
+		// * 20 points to the block producer for producing a block in the chain
 		fn note_author(author: T::AccountId) {
 			let now = <Round<T>>::get().current;
 			let score_plus_20 = <AwardedPts<T>>::get(now, &author).saturating_add(20);
 			<AwardedPts<T>>::insert(now, author, score_plus_20);
 			<Points<T>>::mutate(now, |x| *x = x.saturating_add(20));
 		}
+
+		fn note_uncle(_author: T::AccountId, _age: T::BlockNumber) {}
```

Comment:
why is this a stub? Maybe even add a comment saying why. Same for the others down below.

Reply:
I have to define these methods in order to satisfy the trait. In this case, the `pallet_authorship::EventHandler` trait needs to be able to handle the `note_uncle` event.
For our current implementation, we don't care about a block is an uncle so this is a noop.

For reference, here's the compilation error if the stubbed methods are removed:
```
error[E0046]: not all trait items implemented, missing: `note_uncle`
    --> /Users/nikhil/oak/moonbeam/pallets/parachain-staking/src/lib.rs:1661:2
     |
1661 |     impl<T: Config> pallet_authorship::EventHandler<T::AccountId, T::BlockNumber> for Pallet<T> {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `note_uncle` in implementation
     |
     = help: implement the missing item: `fn note_uncle(_: Author, _: BlockNumber) { todo!() }`

error[E0046]: not all trait items implemented, missing: `end_session`, `start_session`
    --> /Users/nikhil/oak/moonbeam/pallets/parachain-staking/src/lib.rs:1672:2
     |
1672 |     impl<T: Config> pallet_session::SessionManager<T::AccountId> for Pallet<T> {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `end_session`, `start_session` in implementation
     |
     = help: implement the missing item: `fn end_session(_: u32) { todo!() }`
     = help: implement the missing item: `fn start_session(_: u32) { todo!() }`
```

Happy to add a comment if you think that it'll be useful, but I'm not sure what I'd say other than something like "we don't currently care about uncles" which seems less than useful.

Reply:
Just seems weird that we need to fulfill a trait but don't need all the fcns in it. In theory they exist for a reason

Reply:
If it makes you feel better, even the implementation of this trait in cumulus ignores uncles- https://github.com/paritytech/cumulus/blob/9aead492b71990aa6f7bfc0cd35108078f385ae6/pallets/collator-selection/src/lib.rs#L491-L493

Reply:
Same for the session manager trait- https://github.com/paritytech/cumulus/blob/9aead492b71990aa6f7bfc0cd35108078f385ae6/pallets/collator-selection/src/lib.rs#L517-L522

Reply:
yes and no haha